data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwarderConfiguration.java

```diff
@@ -421,4 +429,13 @@ private void validateSslAndAuthentication() {
         if(authentication == ForwardingAuthentication.MUTUAL_TLS && !ssl)
             throw new IllegalArgumentException("Mutual TLS is only available when SSL is enabled.");
     }
+
+    private void setDrainTimeout(final Duration drainTimeout) {
+        if (drainTimeout != null) {
```

Comment:
We should configure the default drain timeout here. That way, callers know they will get a drain timeout and we can keep the defaults here in this one class.

Reply:
Yes we can. I don't think it allows us to get rid of the default defined in the `PipelineParser` though, see above reply to Asif